Assets/Scripts/DataTypes/RouteProbabilities.cs

```diff
@@ -0,0 +1,59 @@
+using System.Linq;
+using System.Collections.Generic;
+using System;
+
+namespace DataTypes
+{
+    public class RouteProbabilities
+    {
+        private int[] _weights;
+        private List<IVertex> _destinations;
+        private List<double> _cumulative;
+
+        public RouteProbabilities(int[] weights, List<IVertex> destinations)
+        {
+            _weights = weights;
+            _destinations = destinations;
+            
+            CalculateCumulative();
+        }
+
+        // modified from: https://stackoverflow.com/a/43345968
+        private void CalculateCumulative()
+        {
+            var sumWeights = (double) _weights.Sum();
+            // generate distribution probability
+            var distribution = _weights.Select(v => v / sumWeights).ToArray();
+
+            var sum = 0d;
+            // first change shape of your distribution probability array
+            // we need it to be cumulative, that is:
+            // if you have [0.1, 0.2, 0.3, 0.4] 
+            // we need     [0.1, 0.3, 0.6, 1  ] instead
+            _cumulative = distribution.Select(c => {
+                var result = c + sum;
+                sum += c;
+                return result;
+            }).ToList();
+        }
+
+        // selects an IVertex from destinations based on weights
+        // modified from: https://stackoverflow.com/a/43345968
+        public IVertex Choose()
+        {
+            var random = new Random();
```

Comment:
vielleicht nen globales random Objekt nutzen: https://stackoverflow.com/questions/2643421/correct-method-of-a-static-random-next-in-c

Reply:
37c00bc